agent/internal/fluent_intg_test.go

```diff
@@ -0,0 +1,258 @@
+//+build integration
+
+package internal
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/google/uuid"
+	"github.com/labstack/echo"
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	aproto "github.com/determined-ai/determined/master/pkg/agent"
+	"github.com/determined-ai/determined/master/pkg/archive"
+	cproto "github.com/determined-ai/determined/master/pkg/container"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/test/testutils"
+)
+
+func TestFluentPostgresLogging(t *testing.T) {
+	// GIVEN a mock master that accepts logs
+	logBuffer, cl := mockLogAcceptor(t)
+	defer cl()
+
+	// AND a successfully started fluent actor
+	aConf := defaultAgentConfig()
+	f, err := newFluentActor(aConf, aproto.MasterSetAgentOptions{
+		MasterInfo: aproto.MasterInfo{},
+		LoggingOptions: model.LoggingConfig{
+			DefaultLoggingConfig: &model.DefaultLoggingConfig{},
+		},
+	})
+	assert.NilError(t, err, "error starting fluentbit")
+	sys := actor.NewSystem("")
+	sys.MustActorOf(actor.Addr("fluent"), f)
+
+	// WHEN a container prints some predefined logs
+	trialID := math.MaxInt16 + rand.Int31n(math.MaxInt16)
+	expected, actual := makeLogTestCase(int(trialID), aConf.AgentID)
+	runContainerWithLogs(t, actual, int(trialID), f.port)
+
+	// THEN fluent should parse all fields as expected and ship them to the mock master.
+	i := 0
+	for {
+		select {
+		case l := <-logBuffer:
+			assertLogEquals(t, l, expected[i])
+			i++
+			if i == len(expected) {
+				return
+			}
+		case <-time.After(time.Minute):
+			assert.Equal(t, i, len(expected), "not enough logs received after one minute")
+		}
+	}
+}
+
+func TestFluentLoggingElastic(t *testing.T) {
+	// GIVEN an elastic instance to accept logs
+	elastic, err := testutils.ResolveElastic()
+	assert.NilError(t, err, "unable to connect to master")
+
+	// AND a successfully started fluent actor
+	aConf := defaultAgentConfig()
+	f, err := newFluentActor(aConf, aproto.MasterSetAgentOptions{
+		MasterInfo:     aproto.MasterInfo{},
+		LoggingOptions: testutils.DefaultElasticConfig(),
+	})
+	assert.NilError(t, err, "error starting fluentbit")
+	sys := actor.NewSystem("")
+	sys.MustActorOf(actor.Addr("fluent"), f)
+
+	// WHEN a container prints some predefined logs
+	trialID := math.MaxInt16 + rand.Int31n(math.MaxInt16)
+	expected, actual := makeLogTestCase(int(trialID), aConf.AgentID)
+	runContainerWithLogs(t, actual, int(trialID), f.port)
+
+	// This is really unfortunate, but we don't query for logs until they're more than 10 seconds old
+	// to _try_ to avoid the trickiness involved with elastic's consistency model.
+	time.Sleep(10 * time.Second)
```

Comment:
The only way to really avoid this is to make it configurable in the master, I can tackle it in a followup if you think it's necessary.